pkg/descheduler/evictions/evictions_test.go

```diff
@@ -40,35 +45,57 @@ func TestEvictPod(t *testing.T) {
 	tests := []struct {
 		description string
 		node        *v1.Node
-		pod         *v1.Pod
-		pods        []v1.Pod
-		want        error
+		evictedPod  *v1.Pod
+		pods        []runtime.Object
+		wantErr     bool
+		wantErrMsg  string
 	}{
 		{
 			description: "test pod eviction - pod present",
 			node:        node1,
-			pod:         pod1,
-			pods:        []v1.Pod{*pod1},
-			want:        nil,
+			evictedPod:  pod1,
+			pods:        []runtime.Object{pod1},
+			wantErr:     false,
 		},
 		{
-			description: "test pod eviction - pod absent",
+			description: "test pod eviction - pod absent (not found error)",
 			node:        node1,
-			pod:         pod1,
-			pods:        []v1.Pod{*test.BuildTestPod("p2", 400, 0, "node1", nil), *test.BuildTestPod("p3", 450, 0, "node1", nil)},
-			want:        nil,
+			evictedPod:  pod1,
+			pods:        []runtime.Object{test.BuildTestPod("p2", 400, 0, "node1", nil), test.BuildTestPod("p3", 450, 0, "node1", nil)},
+			wantErr:     true,
+			wantErrMsg:  fmt.Sprintf("pod not found when evicting \"%s\": pods \"%s\" not found", pod1.Name, pod1.Name),
+		},
+		{
+			description: "test pod eviction - pod absent (too many requests error)",
+			node:        node1,
+			evictedPod:  pod1,
+			pods:        []runtime.Object{test.BuildTestPod("p2", 400, 0, "node1", nil), test.BuildTestPod("p3", 450, 0, "node1", nil)},
+			wantErr:     true,
+			wantErrMsg:  fmt.Sprintf("error when evicting pod (ignoring) \"%s\": Too many requests: too many requests", pod1.Name),
 		},
 	}
 
 	for _, test := range tests {
-		fakeClient := &fake.Clientset{}
-		fakeClient.Fake.AddReactor("list", "pods", func(action core.Action) (bool, runtime.Object, error) {
-			return true, &v1.PodList{Items: test.pods}, nil
+		t.Run(test.description, func(t *testing.T) {
+			fakeClient := fake.NewClientset(test.pods...)
+			fakeClient.PrependReactor("create", "pods/eviction", func(action core.Action) (handled bool, ret runtime.Object, err error) {
+				if test.wantErrMsg == fmt.Sprintf("pod not found when evicting \"%s\": pods \"%s\" not found", test.evictedPod.Name, test.evictedPod.Name) {
```

Comment:
Worth turning the text into a constant given it's used on multiple places. To avoid unintentionally changing the text. E.g.
```go
const (
	notFoundText = "pod not found when evicting \"%s\": pods \"%s\" not found"
	tooManyRequests = "error when evicting pod (ignoring) \"%s\": Too many requests: too many requests"
)
```

Reply:
done